project/src/test/java/com/google/sps/servlets/QueryServletTest.java

```diff
@@ -15,6 +15,7 @@
 package com.google.sps.servlets;
```

Comment:
I'm missing multiple tests in this file:
1. Test that the parameters given in the request are actually the parameters that are inserted into the UserPreferences struct. 
2. Test error handling: what happens if the fetcher throws an exception? the UserPreferencesBuilder? The sorter? JSON parsing error? All of those should be tested (separately!).
3. Test that filtering was actually performed according to the given values. The filter method is static, so you cannot mock it, but you should know what are its results - so make sure it filters correctly.

Reply:
Added a few tests that I hope will answer the requirements. 
About the first test - I'm not sure how such a test is supposed to look, it would be great if we could discuss that

Reply:
Instead of calling 
`when(FETCHER.fetch(any(UserPreferences.class))).`
you need to build the expected UserPreferences object that your class should build given the input you give it in the test, and see that everything is populated correctly. 

Something like 
```java
UserPreferences expectedUP = UserPreferences.builder().setFoo(...)...build();
when(FETCHER.fetch(expectedUp)).then(...); 
```
And then use `verify` to verify the call has actually happened. 

Another option is still use the `any(UserFeedback.class)` and only test the `expectedUP` was provided by calling `verify` on your mock later.

Reply:
Added the test, thanks for the help. 
As we discussed, it will be tested when we refactor the float to be a double, because the float is unstable.

Reply:
👍